File: Golang/ReverseLinkedListII/ReverseLinkedListII_test.go
package ReverseLinkedListII

import (
	"testing"
	"reflect"
)

func TestReverseBetween(t *testing.T) {

	if reverseBetween(nil, 1, 2) == nil {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(reverseBetween(&ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 2,
			Next: &ListNode{
				Val: 3,
			},
		},
	}, 2, 3), &ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 3,
			Next: &ListNode{
				Val: 2,
			},
		},
	}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(reverseBetween(&ListNode{
		Val: 1,
	}, 1, 1), &ListNode{
		Val: 1,
	}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(reverseBetween(&ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 2,
		},
	}, 1, 2), &ListNode{
		Val: 2,
		Next: &ListNode{
			Val: 1,
		},
	}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(reverseBetween(&ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 2,
			Next: &ListNode{
				Val: 3,
				Next: &ListNode{
					Val: 4,
					Next: &ListNode{
						Val: 5,
					},
				},
			},
		},
	}, 2, 4), &ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 4,
			Next: &ListNode{
				Val: 3,
				Next: &ListNode{
					Val: 2,
					Next: &ListNode{
						Val: 5,
					},
				},
			},
		},
	}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}
	
}

File: Python/ReverseLinkedListIITest.py
from unittest import TestCase

from ReverseLinkedListII import ReverseLinkedListII, ListNode


class TestReverseLinkedListII(TestCase):
    def test_reverseBetween(self):
        r = ReverseLinkedListII()

        self.assertIsNone(r.reverseBetween(None, 1, 2))

        node123 = ListNode(1)
        node123.next = ListNode(2)
        node123.next.next = ListNode(3)
        tmp = ListNode(1)
        tmp.next = ListNode(3)
        tmp.next.next = ListNode(2)
        self.assertEqual(r.reverseBetween(node123, 2, 3), tmp)

        tmp = ListNode(1)
        self.assertEqual(r.reverseBetween(ListNode(1), 1, 1), tmp)

        node12 = ListNode(1)
        node12.next = ListNode(2)
        tmp = ListNode(2)
        tmp.next = ListNode(1)
        self.assertEqual(r.reverseBetween(node12, 1, 2), tmp)

        node12345 = ListNode(1)
        node12345.next = ListNode(2)
        node12345.next.next = ListNode(3)
        node12345.next.next.next = ListNode(4)
        node12345.next.next.next.next = ListNode(5)
        tmp = ListNode(1)
        tmp.next = ListNode(4)
        tmp.next.next = ListNode(3)
        tmp.next.next.next = ListNode(2)
        tmp.next.next.next.next = ListNode(5)
        self.assertEqual(r.reverseBetween(node12345, 2, 4), tmp)
